### PR TITLE
Naming style: prefer singular names

### PR DESCRIPTION
**Pull request by ddbeck**
Inspired by https://github.com/web-platform-dx/web-features/pull/1024

### CODE REVIEW

**Comment by ddbeck** [comment]
Already, I think this needs more refinement. It's more specific, perhaps "prefer singular when referring to constructable object types."

**Comment by foolip** [comment]
I think this needs to be qualified in some way, because we have lots of plural names which could be written in singular form. Some would just be wrong because there's more than one, but for a lot of them it's not clear (yet) what the rule is:

- 2D transforms
- 3D transforms
- Array iteration methods (more than one)
- Array iterators
- Async functions
- Audio and video tracks (more than one)
- Border images
- CSS module scripts
- Canvas text baselines (more than one)
- Canvas text metrics (more than one)
- Cascade layers
- Classes
- Compression streams
- Conic gradients
- Constructed stylesheets
- Container queries
- Container style queries
- Custom elements
- Custom highlights
- Custom properties
- Device orientation events (more than one)
- Fetch metadata request headers (more than one)
- File and directory entries
- Form-associated custom elements
- Import maps
- Individual transform properties (more than one)
- Inert elements
- JavaScript modules
- Locks
- Media element pseudo-classes (more than one)
- Mutually exclusive <details> elements
- Notifications
- Push messages
- Registered custom properties
- Relative colors
- Scroll-driven animations
- Set methods (more than one)
- Small, large, and dynamic viewport units (more than one)
- Storage buckets
- Streams
- Trusted types
- Typed array iteration methods (more than one)
- Typed array iterators
- View transitions
- Viewport units
- Web animations
- WebVTT regions
- calc() constants (more than one)

**Comment by ddbeck** [comment]
Yeah, I think the actual rule is something like:

If you're shadowing the literal name of an interface, element, property, etc., don't pluralize it. But if you're referring to the abstract kind of noun, _do_ pluralize.

Or to put it another way: if you could put the name in `<code>` and have it mean the same thing, don't pluralize.

So that's how you get "JavaScript modules" but not "\<dialog\>s" or "BigInts".

That means we've got a few that would need renaming:

- Audio session → Audio sessions
- Managed media source → Managed media sources
- Media session → Media sessions
- Media source → Media sources
- Motion path → Motion paths [Maybe, not quite sure if "Motion path" is a conceptual unit, or if it's paths-that-have-motion.]
- input event → input (event) [This one is special. Other events we just have the name of the literal event, like "scrollend". Here we have "event" so it's not impossibly vague. But that's just disambiguation text, not really part of the name of the feature.)

**Comment by ddbeck** [comment]
I pushed changes to include the required changes to existing features, if this is a route we want to go down.

File: features/audio-sessions.yml [hunk 1]
```diff
@@ -1,4 +1,5 @@
-name: Audio session
+name: Audio sessions
```
[the remark this answers is not in the document]
Comment (ddbeck) [reply features/audio-sessions.yml:1]: When I wrote shadowing I meant in the sense of using "AudioSession" (shadowing) as the name, not turning into a noun phrase "Audio session".

That said, renaming this to "AudioSession" _is_ an option.

File: features/managed-media-sources.yml [hunk 2]
```diff
@@ -1,4 +1,5 @@
-name: Managed media source
+name: Managed media sources
```
[the remark this answers is not in the document]
Comment (ddbeck) [reply features/managed-media-sources.yml:1]: Yeah, same here. An option is to shadow with the name "ManagedMediaSource" (though it suggests an uglier ID, `managedmediasource`).

File: features/audio-sessions.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (foolip) [reply features/audio-sessions.yml:1]: I see, maybe another word than "shadowing" in the guideline would help. Just "matching"?

The question of when to match the interface would remain. Should "Offscreen canvas" be changed to "Offscreen canvases" or renamed to "OffscreenCanvas"?

File: features/managed-media-sources.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (foolip) [reply features/managed-media-sources.yml:1]: The principle I'm gleaning is that if making it plural makes it sound wrong, then the noun phrase was probably wrong all along? That sounds reasonable in the abstract. But I'm worried that the guideline will lead to some cases that feel wrong. Do the changes account for every name with a space in it where a plural/singular decision should be made?

**Comment by ddbeck** [comment]
Philip and talked about this yesterday. After some discussion, we decided that a special rule like this is too difficult to remember and apply. I'm dropping this, but taking up the useful edit here in https://github.com/web-platform-dx/web-features/pull/1492.

**Pull request closed without merging**